### DISCUSSION TITLE
Compatibilidade do checkout bricks com Safari

### DISCUSSION THREAD

**Discussion by FranciscoMaik** [mercadopago/sdk-js#159] [Q&A] [answered]
Olá, há alguma incompatibilidade para montar o formulário de pagamento em alguma versão do Safari? Temos a implementação em um de nossos sistemas e um cliente relatou problemas, ele nos enviou um vídeo mostrando todo o processo e para ele está acontecendo um problema de montagem, desde já agradeço!

**Comment by AdaltonLeite** [comment]
Olá @FranciscoMaik, 

Oficialmente o Brick e a nossa SDK de JS da suporte as versões do Safari 14 ou superiores. Você sabe qual versão do Safari o cliente está usando? Poderia enviar algum evidências do erro acontecendo para a gente tentar simular?

**Comment by FranciscoMaik** [comment]
@AdaltonLeite eu consegui verificar aqui o seguinte a versão do **Safari é o 16.5**;

![WhatsApp Video 2023-06-06 at 12 28 16 (online-video-cutter com)](https://github.com/mercadopago/sdk-js/assets/20601076/6940d6d9-4627-46f9-8acd-70f0522057d6)

Fora isso eu não consigo as informações mais detalhadas.

**Reply by AdaltonLeite** [reply]
@FranciscoMaik qual foi o fluxo para você simular esse erro? Testamos aqui usando o site de vocês e em todas as versões do Safari e do iOS o Brick foi renderizado. Teria um fluxo específico?

**Reply by FranciscoMaik** [reply]
@AdaltonLeite Escolhendo somente o plano da plataforma e indo para o pagamento com o cartão.

**Reply by AdaltonLeite** [reply]
@FranciscoMaik fiz alguns teste aqui usando esse fluxo e sempre apresenta o brick de pagamento. Você consegue simular esse erro do seu lado? É possível ver o inspector se ele exibe algum erro?

Dos teste que realizamos, desde a versão 14.6 até a 16.5 do Safari o Brick tem sido exibido. 


## Exemplos de teste de fizemos: 

### Safari 14.6
<img width="555" alt="Captura de Tela 2023-06-06 às 17 19 53" src="https://github.com/mercadopago/sdk-js/assets/16780687/7e65f2cb-301a-41df-bf66-2212c7ea81de">


### Safari 16.4
<img width="453" alt="Captura de Tela 2023-06-06 às 17 29 15" src="https://github.com/mercadopago/sdk-js/assets/16780687/abf7b83c-8b74-4da5-b30d-2e5ac2836aa4">

**Reply by FranciscoMaik** [reply]
Então pode ter sido um caso muito atípico, fico grato por ter respondido minha dúvida inicial sobre a compatibilidade. Eu não vou conseguir reproduzir devido a não poder ter um contato direto com o cliente, mas estressarei a plataforma para ver se consigo algo semelhante, mas desde já agradeço bastante, já sanou a minha dúvida!

**Reply by AdaltonLeite** [reply, marked as answer]
@FranciscoMaik consegui simular esse cenário no Chrome, me parece se alguma coisa no fluxo de inicialização do pagamento. 

Acredito que ele ta tentando renderizar o Brick antes de atualizar alguma propriedade interna do seu projeto, se o usuário click no botão de **Cartão de crédito** antes do modal de **Aguarde enquanto validamos seus dados** finalizar, ele da um erro de inicialização no brick. Uma sugestão seria desabilitar o click no botão enquanto esse modal não der o status de finalizado, assim você garante que tudo vai tá certo, o que acha? 


Vídeo do exemplo abaixo: 

https://github.com/mercadopago/sdk-js/assets/16780687/b63f2c2c-3f3e-4d1e-bfd7-c73b1a78a545


Espero que isso te ajude, se precisar de mais alguma ajuda só enviar aqui pra gente.
